src/main/java/com/minecolonies/coremod/colony/requestsystem/management/manager/StandardRecipeManager.java

```diff
@@ -55,7 +56,7 @@ public IToken checkOrAddRecipe(final IRecipeStorage storage)
     @Override
     public IToken getRecipeId(final IRecipeStorage storage)
     {
-        for(final Map.Entry<IToken, IRecipeStorage> tempStorage: recipes.entrySet())
+        for(final Map.Entry<IToken, IRecipeStorage> tempStorage: new ArrayList<>(recipes.entrySet()))
```

Comment:
Why is this change made?

Reply:
because even with my changes there was the conc exception

Reply:
But this does not fix it.
The constructor still iterates over it. Please remove.